sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -409,6 +443,9 @@ class Dataset[T] private(
   /** Collects the elements to an Array. */
   def collect(): Array[T] = rdd.collect()
 
+  def jcollect(): java.util.List[T] =
```

Comment:
I do want to avoid this `jcollect`, but our `collect` with return type `Array[T]` seems not working at java side, any ideas?